### ISSUE TITLE
Using `EditorSyntaxHighlighter` in `TextEdit` crashes the project.

### ISSUE THREAD

**Issue by kitbdev** [godotengine/godot#84983]
### Godot version

Godot v4.2.rc (ad72de508)

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 (NVIDIA; 31.0.15.3598) - Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 Threads)

### Issue description

The running project crashes without error message if a `TextEdit` (or `CodeEdit`) has `EditorSyntaxHighlighter` set for its `syntax_highlighter`.

Attaching a debugger to the running project says:
```
ERROR: Class 'EditorSyntaxHighlighter' can only be instantiated by editor.
   at: ClassDB::instantiate (core\object\class_db.cpp:364)
core\object\class_db.cpp:364 - Class 'EditorSyntaxHighlighter' can only be instantiated by editor.
ERROR: res://addons/eval_command/run_script.tscn:5 - Parse Error: Can't create sub resource of type: EditorSyntaxHighlighter
   at: ResourceLoaderText::load (scene\resources\resource_format_text.cpp:551)
scene\resources\resource_format_text.cpp:551 - res://addons/eval_command/run_script.tscn:5 - Parse Error: Can't create sub resource of type: EditorSyntaxHighlighter
ERROR: Failed loading resource: res://addons/eval_command/run_script.tscn. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core\io\resource_loader.cpp:275)
Failed loading resource: res://addons/eval_command/run_script.tscn. Make sure resources have been imported by opening the project in the editor at least once.
ERROR: Failed loading scene: res://addons/eval_command/run_script.tscn.
   at: (main\main.cpp:3433)
Failed loading scene: res://addons/eval_command/run_script.tscn.
```

`EditorSyntaxHighlighter` should probably not be available to select.
Or if `EditorSyntaxHighlighter` is useful for plugins or something, there should be an error printed when attempting to use it when not in the editor.

### Steps to reproduce

Add a `TextEdit` or a `CodeEdit`.
Set Highlighting > `syntax_highlighter` to a new `EditorSyntaxHighlighter`.
Run the current Scene.
The Scene tries to load and immediately closes.

### Minimal reproduction project

N/A

**Comment by jsjtxietian** [comment]
Yes comfirmed. The error message is pretty clear, what should we do about this, at least it shouldn't crash.

**Comment by AThousandShips** [comment]
This doesn't crash, it fails to load the main scene, which is unrecoverable, it happens when you can't load other classes as well, I'd say it's to be expected, how would we gracefully recover from this otherwise?

**Comment by kitbdev** [comment]
It makes sense that it doesn't work, but I think there should be an error message in the Output so it is obvious why.

**Comment by AThousandShips** [comment]
There is? It says it can't instance the scene

**Comment by kitbdev** [comment]
Only when there is a console on the running project, nothing is shown in the editor.

**Comment by AThousandShips** [comment]
That's unrelated to this, that's because the debugger hasn't had a chance to connect yet, that's tracked elsewhere, can't remember the actual issue right now

**Comment by nezvers** [comment]
If `EditorSyntaxHighlighter` and `GDscriptSyntaxHighlighter` crash the project WITHOUT any message in output, then expected behavior is for them not to be in inspector dropdown menu.
 
<img width="1982" height="569" alt="Image" src="https://github.com/user-attachments/assets/92c4b521-4025-41ce-a943-531d9692c982" />
